extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -123,3 +127,73 @@ class AdSenseNetworkConfig {
     };
   }
 }
+
+
+/**
+ * @implements {AdNetworkConfigDef}
+ */
+class DoubleclickNetworkConfig {
+  /**
+   * @param {!Element} autoAmpAdsElement
+   */
+  constructor(autoAmpAdsElement) {
+    this.autoAmpAdsElement_ = autoAmpAdsElement;
+  }
+
+  /**
+   * @param {!Window} win
+   */
+  isEnabled(win) {
+    const branch = getAdSenseAmpAutoAdsExpBranch(win);
+    return branch != AdSenseAmpAutoAdsHoldoutBranches.CONTROL;
+  }
+
+  /** @override */
+  getConfigUrl() {
+    const docInfo = Services.documentInfoForDoc(this.autoAmpAdsElement_);
+    const canonicalHostname = parseUrlDeprecated(docInfo.canonicalUrl).hostname;
+    return buildUrl('//pagead2.googlesyndication.com/getconfig/ama', {
+      'client': this.autoAmpAdsElement_.getAttribute('data-ad-legacy-client'),
+      'plah': canonicalHostname,
+      'ama_t': 'amp',
+      'url': docInfo.canonicalUrl,
+    }, 4096);
+  }
+
+  /** @override */
+  getAttributes() {
+    const attributes = dict({
+      'type': 'doubleclick',
+      'data-slot': this.autoAmpAdsElement_.getAttribute('data-slot'),
+      'json': this.autoAmpAdsElement_.getAttribute('data-json'),
+    });
+    const experimentJson = tryParseJson(
+        this.autoAmpAdsElement_.getAttribute('data-experiment'));
+    if (experimentJson) {
+      if (experimentJson['height']) {
+        attributes['height'] = experimentJson['height'];
+      }
+      if (experimentJson['width']) {
+        attributes['width'] = experimentJson['width'];
+      }
+      if (experimentJson['layout']) {
+        attributes['layout'] = experimentJson['layout'];
```

Comment:
It seems like this circumvents the AMP validator. In theory only layouts that are whitelisted are allowed for given elements and if a non-whitelisted one is set then the AMP validator would give an error. Unless we modified the validator to validate the contents of the experiment JSON, then this introduces a mechanism by which people can set invalid layouts.